src/variables.jl

```diff
@@ -807,6 +814,46 @@ function _moi_constrain_variable(backend::MOI.ModelLike, index, info)
     end
 end
 
+"""
+    ConstrainedVariables <: AbstractVariable
```

Comment:
We should also propagate the new naming here, e.g., `VariableConstrainedOnCreation`.